File: lib/oxford_comma.js
function oxford_comma(array) {
  if (array.length >= 3) {
    var lastEle = array[array.length - 1];
    var firstHalf = array.slice(0, array.length - 1);

    return firstHalf.join(", ") + ", and " + lastEle;
  } else if (array.length === 2) {
    return array.join(" and ");
  } else {
    return array.join("");
  }
}

File: lib/oxford_comma.rb
def oxford_comma(array)
  if array.size == 1
    return array.join()
  elsif array.size == 2
    return array.join(" and ")
  elsif array.size >= 3
    last_ele = array.slice(-1)
    remainding_ele = array.slice(0, array.size - 1)
    
    return remainding_ele.join(", ") + ", and " + last_ele
  else
    return array
  end
end
